crates/bin/pcli/src/transaction_view_ext.rs

```diff
@@ -0,0 +1,349 @@
+                                    swap_plaintext.trading_pair.asset_2(),
+                                ),
+                                // The pathological case (both assets have output values).
+                                _ => (
+                                    swap_plaintext.trading_pair.asset_1(),
+                                    swap_plaintext.delta_1_i,
+                                    swap_plaintext.trading_pair.asset_1(),
+                                )
+                            };
+
+                            action = format!(
+                                "{} {} for {} and paid claim fee {:?}",
+                                from_value,
+                                from_asset,
+                                to_asset,
+                                format_fee_view(&swap_plaintext.claim_fee),
+                            );
+                            
+                            ["Swap", &action]
+                        }
+                        SwapView::Opaque { swap } => {
+                            action = format!(
+                                "Opaque swap for trading pair: {} <=> {}",
+                                swap.body.trading_pair.asset_1(),
+                                swap.body.trading_pair.asset_2()
+                            );
+                            ["Swap", &action]
+                        }
+                    }
+                }
+                penumbra_transaction::ActionView::SwapClaim(swap_claim) => {
+                    match swap_claim {
+                        SwapClaimView::Visible {
+                            swap_claim,
+                            output_1,
+                            output_2,
+                        } => {
+                            dbg!(swap_claim); // View service can't see SwapClaims?
+                            let claimed_value = match (value_view_amount(&output_1.value).value(), value_view_amount(&output_2.value).value()) {
+                                (0, v) if v > 0 => format_value_view(&output_2.value),
+                                (v, 0) if v > 0 => format_value_view(&output_1.value),
+                                // The pathological case (both assets have output values).
+                                _ => format!(
+                                    "{} and {}",
+                                    format_value_view(&output_1.value),
+                                    format_value_view(&output_2.value),
+                                ),
+                            };
+                            
+                            action = format!(
+                                "Claimed {} with fee {:?}",
+                                claimed_value,
+                                format_fee_view(&swap_claim.body.fee),
+                            );
+                            ["Swap Claim", &action]
+                        }
```

Comment:
Was unable to test the SwapView::Visible match since the view service could never find transactions containing the swaps I made. When fetching from a fullnode, I don't expect to get a SwapView::Visible, but do expect the view service to find the transaction I just executed. Not sure why this is happening.

Reply:
It could be related to a long-running bug that we have not got around to fixing yet: https://github.com/penumbra-zone/penumbra/issues/2547

So probably not important to go to great lengths to exercise in this PR.